tests/returntypes/test_vjp_new.py

```diff
@@ -0,0 +1,665 @@
+    def test_dtype_matches_dy(self, dtype):
+        """Tests that the vjp function matches the dtype of dy when dy is
+        zero-like."""
+        x = np.array([0.1], dtype=np.float64)
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(x[0], wires=0)
+            qml.expval(qml.PauliZ(0))
+
+        dy = np.zeros(3, dtype=dtype)
+        _, func = qml.gradients.vjp(tape, dy, qml.gradients.param_shift)
+
+        assert func([]).dtype == dtype
+
+
+def expected(params):
+    x, y = 1.0 * params
+    return (
+        np.array(
+            [
+                (np.cos(y / 2) ** 2 * np.sin(x)) + (np.cos(y / 2) ** 2 * np.sin(x)),
+                (np.cos(x / 2) ** 2 * np.sin(y)) - (np.sin(x / 2) ** 2 * np.sin(y)),
+            ]
+        )
+        / 2
+    )
+
+
+def ansatz(x, y):
+    qml.RX(x, wires=[0])
+    qml.RY(y, wires=[1])
+    qml.CNOT(wires=[0, 1])
+    qml.probs(wires=[0, 1])
+
+
+class TestVJPGradients:
+    """Gradient tests for the vjp function"""
+
+    @pytest.mark.autograd
+    def test_autograd(self, tol):
+        """Tests that the output of the VJP transform
+        can be differentiated using autograd."""
+        dev = qml.device("default.qubit.autograd", wires=2)
+        params = np.array([0.543, -0.654], requires_grad=True)
+
+        def cost_fn(x, dy):
+            with qml.tape.QuantumTape() as tape:
+                ansatz(x[0], x[1])
+
+            tape.trainable_params = {0, 1}
+            tapes, fn = qml.gradients.vjp(tape, dy, param_shift)
+            vjp = fn(dev.batch_execute(tapes))
+            return vjp
+
+        dy = np.array([-1.0, 0.0, 0.0, 1.0], requires_grad=False)
+        res = cost_fn(params, dy)
+        assert np.allclose(res, expected(params), atol=tol, rtol=0)
+
+        res = qml.jacobian(cost_fn)(params, dy)
+        assert np.allclose(res, qml.jacobian(expected)(params), atol=tol, rtol=0)
+
+    @pytest.mark.torch
+    def test_torch(self, tol):
+        """Tests that the output of the VJP transform
+        can be differentiated using Torch."""
+        import torch
+
+        dev = qml.device("default.qubit", wires=2)
+
+        params_np = np.array([0.543, -0.654], requires_grad=True)
+        params = torch.tensor(params_np, requires_grad=True, dtype=torch.float64)
+        dy = torch.tensor([-1.0, 0.0, 0.0, 1.0], dtype=torch.float64)
+
+        with qml.tape.QuantumTape() as tape:
+            ansatz(params[0], params[1])
+
+        tape.trainable_params = {0, 1}
+        tapes, fn = qml.gradients.vjp(tape, dy, param_shift)
+        vjp = fn(qml.execute(tapes, dev, qml.gradients.param_shift, interface="torch"))
+        print(vjp)
+        assert np.allclose(vjp.detach(), expected(params.detach()), atol=tol, rtol=0)
+
+        cost = vjp[0]
+        cost.backward()
+
+        exp = qml.jacobian(lambda x: expected(x)[0])(params_np)
+        assert np.allclose(params.grad, exp, atol=tol, rtol=0)
+
+    @pytest.mark.tf
+    @pytest.mark.slow
```

Comment:
In general: are there any more slow tests to mark that have been added in this PR?

Reply:
They are some, but they are just the duplicate from the old test suite.